Remotion/ObjectBinding/Web.Development.WebTesting/ControlObjects/BocControlObject.cs

```diff
@@ -66,6 +66,25 @@ protected IReadOnlyList<string> GetValidationErrors ([NotNull] ElementScope scop
       if (IsReadOnly())
         throw AssertionExceptionUtility.CreateControlReadOnlyException();
 
+      return GetValidationErrorsInner (scope);
+    }
+
+    /// <summary>
+    /// Returns the validation errors for the read-only <param name="scope" />.
+    /// </summary>
+    /// <exception cref="MissingHtmlException">Thrown if the validation errors cannot be found due to faulty markup or missing validator.</exception>
+    protected IReadOnlyList<string> GetValidationErrorsForReadOnly ([NotNull] ElementScope scope)
+    {
+      ArgumentUtility.CheckNotNull ("scope", scope);
+
+      if (!IsReadOnly())
+        throw new MissingHtmlException ("The control is currently not in a read-only state. Therefore, the operation is not possible.");

```

Comment:
should we change this to AssertionExceptionUtility.CreateControlNotReadOnlyExcception ? Also, exception type.